### PR TITLE
feat: add translate animation

### PR DESCRIPTION
**Pull request by Blakko**
##### Checklist
- [x] `npm test` passes
- [x] commit message follows commit guidelines

##### Description of change
This adds an option to perform an animated translate.
I added this to allow performing an animated padding, example below (implemented with graphin)

https://user-images.githubusercontent.com/2861371/142659666-d74740d2-8c7c-44c0-85ba-765d0f9bdbb0.mp4

### CODE REVIEW

File: packages/core/src/graph/graph.ts [hunk 1]
```diff
@@ -556,26 +556,71 @@ export default abstract class AbstractGraph extends EventEmitter implements IAbs
...
 
-    group.setMatrix(matrix);
+      var dx_1 = dx * matrix[0];
+      var dy_1 = dy * matrix[4];
+
+      var lastX_1 = 0;
+      var lastY_1 = 0;
+      var newX_1 = 0;
+      var newY_1 = 0;
+      group.animate(function (ratio) {
```
Comment (Yanyan-Wang) [inline packages/core/src/graph/graph.ts:600]: Please use arrow function instead

File: packages/core/src/graph/graph.ts [hunk 2]
```diff
@@ -556,26 +556,71 @@ export default abstract class AbstractGraph extends EventEmitter implements IAbs
...
+          }
+        } else {
+          animateConfig.callback = () => {
+            this.emit('viewportchange', { action: 'translate', matrix });
+          }
+        }
+      }
 
-    group.setMatrix(matrix);
+      var dx_1 = dx * matrix[0];
```
Comment (Yanyan-Wang) [inline packages/core/src/graph/graph.ts:593]: var -> const or let

File: packages/core/src/graph/graph.ts [hunk 3]
```diff
@@ -556,26 +556,71 @@ export default abstract class AbstractGraph extends EventEmitter implements IAbs
...
-  public translate(dx: number, dy: number): void {
+  public translate(dx: number, dy: number, animate?: boolean, animateCfg?: GraphAnimateConfig): void {
     const group: IGroup = this.get('group');
 
     let matrix = clone(group.getMatrix());
     if (!matrix) {
       matrix = [1, 0, 0, 0, 1, 0, 0, 0, 1];
     }
-    matrix = transform(matrix, [['t', dx, dy]]);
+    if (animate) {
```
Comment (Yanyan-Wang) [inline packages/core/src/graph/graph.ts:569]: Is it possible to extract a util function for the translate animate and zoom animate? Line 570 - Line 607

File: packages/core/src/graph/graph.ts (hunk 2 above)
Comment (Yanyan-Wang) [reply packages/core/src/graph/graph.ts:593]: and Camel-Case for variable names pls

File: packages/core/src/graph/graph.ts [hunk 4]
```diff
@@ -556,26 +556,71 @@ export default abstract class AbstractGraph extends EventEmitter implements IAbs
    * 平移画布
    * @param dx 水平方向位移
    * @param dy 垂直方向位移
+   * @param {boolean} animate 是否带有动画地移动
```
Comment (Yanyan-Wang) [inline packages/core/src/graph/graph.ts:559]: Wow you do speak Chinese?

File: packages/core/src/graph/graph.ts (hunk 3 above)
Comment (Yanyan-Wang) [reply packages/core/src/graph/graph.ts:569]: Or reuse the `move` like here https://github.com/antvis/G6/blob/b7483c06e82ad5fb8407a2241d62e0574e91e416/packages/core/src/graph/graph.ts#L582

Notice that the second parameter { x, y } for `move` means the target position. Maybe just transform the vector x y of `translate` to the target x y and call `move` with animateCfg

**Comment by Yanyan-Wang** [comment]
Thanks for you Pull Request! Pls checkout the comments.

File: packages/core/src/graph/graph.ts (hunk 4 above)
Comment (Blakko) [reply packages/core/src/graph/graph.ts:559]: Not at all, I simply copied the comment from another function :)

**Comment by Blakko** [comment]
I changed the animated translate to use the move function. Additionally, I refactored the code to ensure that the callback handling behaves the same way between the animated zoom and the animated translate

**Pull request merged**